### ISSUE TITLE
[1.5.5 -> master] no-member for inferred bool types

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#922]
I noticed a couple of issues when trying out astroid/pylint master, with a codebase which is pylint-clean with astroid 1.4.5 and pylint 1.5.5. I'll open an issue for thing I think is separate.

Those can all be reproduced by cloning the [qutebrowser repository](https://github.com/The-Compiler/qutebrowser) and running `tox -e pylint-master -r`. All line numbers are on commit 24db93f3eb17b97e8928d5faf26dd56221030493 of qutebrowser.

---

```
************* Module qutebrowser.config.configtypes
E:1473,24: Instance of 'bool' has no 'items' member (no-member)

************* Module qutebrowser.misc.ipc
E:334,14: Instance of 'bool' has no 'get' member (no-member)
```

This seems due to `bool` being inferred from `json.get`, similar to #753 and some warnings in #754:

``` python
        try:
            json_data = json.loads(decoded)
        except ValueError:
            # [...]
            return

        # [...] 

        cwd = json_data.get('cwd', '')  # <---
```

``` python
        try:
            json_val = json.loads(value)
        except ValueError as e:
            raise configexc.ValidationError(value, str(e))

        if not isinstance(json_val, dict):
            raise configexc.ValidationError(value, "Expected json dict, but "
                                            "got {}".format(type(json_val)))

        # [...]

        for key, val in json_val.items():   # <---
            # [...]
```

Probably the underlying bug is https://github.com/PyCQA/astroid/issues/75 - not sure why they turn up now though - just improved inferrence?

**Comment by hajs** [comment]
The following patch to astroid fixed the issue for me:
Just put the module as `brain_json.py` in `astroid/brain` 

```python
import astroid


def _json_transform():
    # pylint always thinks that json.loads returns a bool. 
    # Message: Instance of 'bool' has no 'get' member (no-member)
    return astroid.parse('''
    def loads(s, encoding=None, cls=None, object_hook=None, parse_float=None,
        parse_int=None, parse_constant=None, object_pairs_hook=None, **kw):
        if 1 in kw: return {}
        if 2 in kw: return []
        if 3 in kw: return True
        if 4 in kw: return ""
        if 5 in kw: return u""
        if 6 in kw: return None
        if 7 in kw: return 3.14
    ''')


astroid.register_module_extender(astroid.MANAGER, 'json', _json_transform)
astroid.register_module_extender(astroid.MANAGER, 'simplejson', _json_transform)
```


Test:
```
import json

demo = '{"foo": 42}'
value = json.loads(demo)
print(value.keys())
print(value.values())
print(len(value))
print(value.items())
print(value.get("foo"))
print(value["foo"])

demo = '[1,2,3]'
value = json.loads(demo)
value.append(4)
print(value[-1])

demo = '"hello"'
value = json.loads(demo)

print(value.capitalize() + ", world")

```

**Comment by PCManticore** [comment]
This looks nice. Care sending it as a pull request to astroid?

**Comment by PCManticore** [comment]
This is fixed in master. Well, technically not fixed, we don't emit no-member any longer when the inference returns opaque inference objects, such as ``Uninferable`` or ``Unknown``. The root cause is that we cannot fully understand most of dynamic Python code, so this check suffers from too many false positives. If you still want the old behaviour, it can be enabled with ``--ignore-on-opaque-inference=n``.
